21 Actor

kosara-g opened the pull request:
Implemented actor class. All diagrams with at least 1 class will have the first class rendered as a king.

Commit:
Reimplemented changes.

Rashadse commented on Application/src/visuals/ActorClass.java:
```diff
@@ -0,0 +1,144 @@
+ * An ActorClass wrapper modeled after DiagramClass.
+ * @author Pontus Laestadius
+ * collaborator Kosara Golemshinska
+ * @version 1.0
```
doesn't the versioning start with 0.1?

kosara-g replied:
good point, I will check with the rest of the code

timjon replied:
It starts at 1 when it gets put up for a pull request if I remember it correctly

kosara-g replied:
oh, okay, thanks

Rashadse commented on Application/src/visuals/ActorClass.java:
```diff
@@ -0,0 +1,144 @@
+
+    /**
+     * From: Renderable
+     * @return nothing, as it's not in use.
```
do we need it if it is not in use?

kosara-g replied:
It's part of the interface, so the compiler complains if they're not implemented

timjon commented on Application/src/visuals/ActorClass.java:
```diff
@@ -0,0 +1,144 @@
+ */
+public class ActorClass implements Renderable {
+
+	private Coordinates coordinates; // The coordinates of the object.
```
Is the indentation weird here or is it just git?

kosara-g replied:
Looks fine for me, I'll fix it and push again

timjon commented on Application/src/visuals/Message.java:
```diff
@@ -167,4 +167,16 @@ public int getToNode(){
+	@Override
+	public String getName() {
+		// Not in use.
+		return null;
```
Shouldnt this be returning something?

kosara-g replied:
Not at the moment, had to implement it as the compiler complained about the interface lacking classes, which automatically made them unimplemented in 2 other classes.

timjon replied:
I think that it should be implemented if it has to be there in case someone has to use it

kosara-g replied:
interface method

timjon replied:
This is in the message class so it should be implemented

timjon commented on Application/src/visuals/Message.java:
```diff
@@ -167,4 +167,16 @@ public int getToNode(){
+
+	@Override
+	public void place(Coordinates coordinates, int size) {
+		// Not in use.
```
Shouldnt this method do something?

kosara-g replied:
It's only used in DiagramClass but since both it and Message implement Renderable, the method needs to be implemented.

timjon replied:
I think that it should be implemented if it has to be there in case someone has to use it

kosara-g replied:
again, part of the interface

timjon replied:
This is in the message class so it should be implemented

timjon commented on Application/src/visuals/Renderable.java:
```diff
@@ -20,4 +20,10 @@
     Coordinates getCoordinates();
+
+    // Returns the name for this object.
+    String getName();
```
Whats the point of this line?

kosara-g replied:
ActorClass implements Renderable, therefore it needs to be defined there. It's then called in Draw.

timjon replied:
I missed that it was an interface my bad

timjon commented on Application/src/visuals/Renderable.java:
```diff
@@ -20,4 +20,10 @@
+    String getName();
+
+    // Places the object at the specified coordinates.
+	void place(Coordinates coordinates, int size);
```
This does nothing right?

kosara-g replied:
It does in DiagramClass. This implementation is then called in Draw.

timjon replied:
I missed that it was an interface my bad

Commit:
Fixed weird indentation

timjon reviewed (changes requested):
Forgot to write this in the previous review:
Isnt the Actorclass file pretty much a copy paste from Pontus diagramclass

kosara-g commented:
@timjon Yes, it says right there in the first lines. Initially, it extended DiagramClass but that was too buggy, so I changed it to a separate class instead. Pontus is still an author :)

Commit:
Updated the actor class

ActorClass now extends DiagramClass, reducing the size in half opposed
to before. Had to change some access modifiers to package-private to
access them in this subclass.

Commit:
Added Override annotation

Commit:
Added a red carpet for the king to stand on.

IsabelleTornqvist reviewed (approved):
Ran through it, worked for me. Looks good.

timjon commented on Application/src/visuals/ActorClass.java:
```diff
@@ -0,0 +1,93 @@
+     * @param gc GraphicalContext for a canvas.
+     */
+    @Override
+    public void render(GraphicsContext gc) {
```
This is still just duplicate code but if everyone else is fine with it then I guess we can have it as technical debt

kosara-g replied:
It isn't though, it differs, so I had to override it. If you look at line 82, I had to change the last 2 arguments to fit the king image onto the platform.

timjon replied:
its 99% the same

kosara-g replied:
so what's your suggestion?

timjon replied:
this: "This is still just duplicate code but if everyone else is fine with it then I guess we can have it as technical debt"

kosara-g replied:
okay, we can refactor next sprint, I suppose

Commit:
Implemented interface methods

kosara-g commented:
@timjon just implemented the Renderable methods
